src/test/obj_fragmentation2/obj_fragmentation2.c

```diff
@@ -57,14 +58,77 @@ remove_last()
 	return obj;
 }
 
+static void
+delete_objects(PMEMobjpool *pop, float pct)
+{
+	size_t nfree = (size_t)(nobjects * pct);
+
+	PMEMoid oid = pmemobj_root(pop, 1);
+
+	shuffle_objects(0, nobjects);
+	while (nfree--) {
+		oid = remove_last();
+
+		allocated_current -= pmemobj_alloc_usable_size(oid);
+
+		pmemobj_free(&oid);
+	}
+}
+
+/*
+ * object_next_size -- generates random sizes in range with
+ *	exponential distribution
+ */
+static size_t
+object_next_size(size_t max, size_t min)
+{
+	float n = rnd64() / (UINT64_MAX / 1.0f);
+	return (max) - (size_t)((max) * sqrt(sqrt(n))) + min;
```

Comment:
This formula doesn't generate values between min and max.

Reply:
You probably want min + (max-min) * exp(-4*n) or min + (max-min) * (1-exp(-4*n))

Reply:
Math is hard :D
It was good enough, but I changed it to your version.